### PR TITLE
WIP - JAMES-3028 extract dumb blob store for object store

### PR DESCRIPTION
**Pull request by remk**

### CODE REVIEW

File: server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageDumbBlobStore.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        return AwsS3ObjectStorage.blobStoreBuilder(testConfig);
+    }
+
+    @PreDestroy
+    public void close() throws IOException {
+        blobStore.getContext().close();
+        blobPutter.close();
+    }
+
+    private boolean isItABigStream(InputStream bufferedData) throws IOException {
```
Comment (chibenwa) [inline server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageDumbBlobStore.java:96]: No longer needed. 

Remove the save(InputSteam) methodfrom the dumb blob store api, and get the final size from the byteSource.

Also, the caller passes the blovId, this code can be simplified compared to the previous version of it.

File: server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageDumbBlobStore.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class ObjectStorageDumbBlobStore implements DumbBlobStore {
+    private static final int BUFFERED_SIZE = 256 * 1024;
+    public static final boolean LAZY = false;
+
+    private final BucketName defaultBucketName;
+    private final org.jclouds.blobstore.BlobStore blobStore;
+    private final BlobPutter blobPutter;
```
Comment (chibenwa) [inline server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageDumbBlobStore.java:59]: Given that we no longer copy stuff we might want to evaluate if keeping the blobPutter still makes sens (separate pr)

**Comment by chibenwa** [comment]
```
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [ERROR] src/main/java/org/apache/james/blob/objectstorage/ObjectStorageDumbBlobStoreBuilder.java:[25,8] (imports) UnusedImports: Unused import - org.apache.james.blob.api.BlobId.
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [ERROR] src/main/java/org/apache/james/blob/objectstorage/ObjectStorageDumbBlobStoreBuilder.java:[30,8] (imports) UnusedImports: Unused import - com.google.common.base.Preconditions.
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [ERROR] src/main/java/org/apache/james/blob/objectstorage/ObjectStorageDumbBlobStore.java:[49] (imports) ImportOrder: 'reactor.core.publisher.Flux' should be separated from previous imports.
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/scalatest/scalatest_2.13/3.1.1/scalatest_2.13-3.1.1.jar (6.9 MB at 15 MB/s)
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [INFO] 
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ blob-deduplicating ---
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [INFO] 
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven) @ blob-deduplicating ---
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [INFO] 
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [INFO] --- git-commit-id-plugin:3.0.1:revision (get-the-git-infos) @ blob-deduplicating ---
[93c78ffb9281d2290d760f4ad51e5701f6d30fa3] [ERROR] src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStore.java:[39] (imports) ImportOrder: 'reactor.core.publisher.Mono' should be separated from previous imports.
```

**Comment by chibenwa** [comment]
```
[47e390d2fca88427c0345c56930c6b8b110346d1] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project blob-objectstorage-guice: Compilation failure: Compilation failure: 
[47e390d2fca88427c0345c56930c6b8b110346d1] [ERROR] /james-project/server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/swift/SwiftObjectStorage.java:[23,43] cannot find symbol
[47e390d2fca88427c0345c56930c6b8b110346d1] [ERROR]   symbol:   class ObjectStorageBlobStoreBuilder
[47e390d2fca88427c0345c56930c6b8b110346d1] [ERROR]   location: package org.apache.james.blob.objectstorage
[47e390d2fca88427c0345c56930c6b8b110346d1] [ERROR] /james-project/server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/swift/SwiftObjectStorage.java:[57,48] package ObjectStorageBlobStoreBuilder does not exist
[47e390d2fca88427c0345c56930c6b8b110346d1] [ERROR] /james-project/server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/swift/SwiftObjectStorage.java:[61,48] package ObjectStorageBlobStoreBuilder does not exist
[47e390d2fca88427c0345c56930c6b8b110346d1] [ERROR] /james-project/server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/swift/SwiftObjectStorage.java:[65,48] package ObjectStorageBlobStoreBuilder does not exist
```

**Comment by blackheaven** [comment]
test this please

**Comment by chibenwa** [comment]
Abandonned.

See https://github.com/linagora/james-project/pull/3773

**Pull request closed without merging**